File: Server_npm/crm.js
const express = require('express');
const router = express.Router();
const axios = require('axios');

// Fonction pour obtenir les données mock
async function getMockData(endpoint) {
  try {
    const response = await axios.get(`https://615f5fb4f7254d0017068109.mockapi.io/api/v1/${endpoint}`);
    if (response.status === 200 && response.data) {
      return response.data;
    } else {
      console.error(`Error fetching data from ${endpoint}`);
      return null;
    }
  } catch (error) {
    console.error(`Error fetching data from ${endpoint}: ${error.message}`);
    return null;
  }
}

// Fonction pour obtenir les données de prospects
function getProspectsData(clientsData) {
  return clientsData.filter(client => !client.orders || client.orders.length === 0);
}

// API pour les clients du CRM
router.get('/clients', async (req, res) => {
  const clientsData = await getMockData('customers');
  res.json(clientsData);
});

// API pour les prospects du CRM
router.get('/prospects', async (req, res) => {
  const clientsData = await getMockData('customers');
  const prospectsData = getProspectsData(clientsData);
  res.json(prospectsData);
});

module.exports = router;

File: backend/crm/crm.py
from flask import Flask, jsonify

from libs.get_mock_api_data.get_mock_api_data import get_mock_data

app = Flask(__name__)


def get_prospects_data(clients_data):
    prospects_data = [client for client in clients_data if "orders" not in client or not client["orders"]]
    return prospects_data


# API pour les clients
@app.route('/api/clients', methods=['GET'])
def get_clients():
    clients_data = get_mock_data("customers")
    return jsonify(clients_data)


# API pour les prospects
@app.route('/api/prospects', methods=['GET'])
def get_prospects():
    clients_data = get_mock_data("customers")
    prospects_data = get_prospects_data(clients_data)
    return jsonify(prospects_data)


clients_data = get_mock_data("customers")
prospects_data = get_prospects_data(clients_data)


# Utilisation des données
if clients_data:
    print("Clients:", clients_data, "\n\n\n")

if prospects_data:
    print("Prospects:", prospects_data, "\n\n\n")